graph-db/src/utils/indexing/indexCmrCollection.js

```diff
@@ -0,0 +1,46 @@
+require('array-foreach-async')
+const { indexRelatedUrl } = require('./indexRelatedUrl')
+
+/**
+ * Given a collection from the CMR, index it into Gremlin
+ * @param {JSON} collection collection object from `items` array in cmr response
+ * @param {Gremlin Traversal Object} gremlin connection to gremlin server
+ * @returns
+ */
+exports.indexCmrCollection = async (collection, gremlin) => {
+  const { meta, umm } = collection
+  const { 'concept-id': conceptId } = meta
+  const { EntryTitle: entryTitle, DOI: doi, RelatedUrls: relatedUrls } = umm
+  const { DOI: doiDescription } = doi
+  let doiUrl = 'Not provided'
+  let datasetName = `${process.env.CMR_ROOT}/concepts/${conceptId}.html`
+
+  if (doiDescription) {
+    // Take the second element from the split method
+    const [, doiAddress] = doiUrl.split(':')
+    doiUrl = `http://doi.org/${doiAddress}`
+    datasetName = doiUrl
+  }
+
+  const exists = await gremlin.V().hasLabel('dataset').has('concept-id', conceptId).hasNext()
+  let dataset = null
+  if (!exists) {
+    dataset = await gremlin
+      .addV('dataset')
```

Comment:
Have we thought about creating a class that creates the entire graph structure that is only run when needed? That way all of the graph structure exists in one place instead of spread out throughout the indexing code.  Once the structure is in place, then we can just index the data.